### PR TITLE
Fixed the entropy calculation of the von Mises distribution.

### PR DESCRIPTION
**Pull request by NeilGirdhar**
(scipy.stats.vonmises)

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -4496,13 +4497,17 @@ def _rvs(self, kappa):
...
 
     def _cdf(self, x, kappa):
         return _stats.von_mises_cdf(kappa, x)
 
     def _stats_skip(self, kappa):
         return 0, None, 0, None
+
+    def _entropy(self, kappa):
+        return (-kappa * i1(kappa) / i0(kappa)
+                + math.log(2 * math.pi * i0(kappa)))
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: math.log is going to break down for array_like kappa. Better use np.log

Comment (NeilGirdhar) [inline scipy/stats/_continuous_distns.py]: Thanks!

**Comment by NeilGirdhar** [comment]
`check_private_entropy` says:

> compare a generic _entropy with the distribution-specific implementation

but the generic entropy is broken.  I have no idea how to fix this.

**Comment by ev-br** [comment]
You'll need to skip the test if `distname =='vonmises'`

**Comment by ev-br** [comment]
Now the last thing would require a bit of wrangling with git, I'm afraid. First, please don't merge master into feature branches. Workflows differ; we typically prefer rebasing on master if necessary. Second, six commits for ten LOC is a bit much (there's no hard rule to limit the number of commits per PR, but here it seems fairly clear). Thus --- would you please squash the commits. While you're at it, it could be a good time to also rewrite the commit message to start with a standard acronym (http://docs.scipy.org/doc/numpy-1.10.1/dev/gitwash/development_workflow.html). Something like "BUG" would be appropriate, I'd think. Thanks!

**Comment by ev-br** [comment]
Squashed and merged in 58b4f54ea77e47b754bcf237f0149e95a5511770.
Thanks @NeilGirdhar and congratulations with the first scipy commit! (6c7ec1d2c7be60a2a3b482eb7f4e889288707ecb)

**Comment by NeilGirdhar** [comment]
@ev-br  Thanks for taking care of the squash.

**Pull request closed without merging**